services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -0,0 +1,390 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flow.service;
+
+import static java.util.Objects.requireNonNull;
+
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPair;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.FlowSegment;
+import org.openkilda.model.Switch;
+import org.openkilda.pce.PathComputer;
+import org.openkilda.pce.PathComputerFactory;
+import org.openkilda.pce.PathComputerFactory.Strategy;
+import org.openkilda.pce.PathPair;
+import org.openkilda.pce.RecoverableException;
+import org.openkilda.pce.UnroutableFlowException;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.FlowSegmentRepository;
+import org.openkilda.persistence.repositories.IslRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchRepository;
+import org.openkilda.wfm.topology.flow.validation.FlowValidationException;
+import org.openkilda.wfm.topology.flow.validation.FlowValidator;
+import org.openkilda.wfm.topology.flow.validation.SwitchValidationException;
+
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+public class FlowService extends BaseFlowService {
+    private SwitchRepository switchRepository;
+    private FlowSegmentRepository flowSegmentRepository;
+    private IslRepository islRepository;
+    private PathComputerFactory pathComputerFactory;
+    private FlowResourcesManager flowResourcesManager;
+    private FlowValidator flowValidator;
+
+    public FlowService(PersistenceManager persistenceManager, PathComputerFactory pathComputerFactory,
+                       FlowResourcesManager flowResourcesManager, FlowValidator flowValidator) {
+        super(persistenceManager);
+        transactionManager = persistenceManager.getTransactionManager();
```

Comment:
You don't need to rewrite it here. `transactionManager` will be initialized in `BaseFlowService`.

Reply:
Fixed.